### PR TITLE
Adds a Pii Flag, Censors Hwid and Ip 

### LINKED ISSUE #6859: Restricting access to user PII behind a +CHECKUSER admin permission.
This would mean obscuring the IP and HWID of the user in some non-reversible manner or just hiding it outright, and not allowing non-checkuser staff to apply IP bans (and simply presenting a "sticky ban" button that does so but with the details obscured).
This requires changes to SS14.Admin as well.

### PR DESCRIPTION
**Pull request by Geekyhobo**
## About the PR

Add a Pii Flag to Admin Flags, Closes #6859, Unfucked version of #23720

## Why / Balance

Helpful for restricting Pii behind permission, not everyone should have access to that type of sensitive info, also this has much greater use on SS14.Admin where there is PII all over the place, with a flag I can now hide that Pii (Coming SoonTM)

## Technical details

Censors Pii to "*Hidden\*" based on an admin flag

## Media

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

Screenshot 2024-05-07 140143

## Breaking changes

**Changelog**

:cl:
ADMIN:
- add: Added a PII admin flag to hide personal identification information.

### CODE REVIEW

File: Content.Server/Administration/BanList/BanListEui.cs [hunk 1]
```diff
@@ -65,21 +65,40 @@ private async Task LoadBans(NetUserId userId)
...
+                    ban.HWId == null ? null : Convert.ToBase64String(ban.HWId.Value.AsSpan()),
+                    ban.BanTime.UtcDateTime,
+                    ban.ExpirationTime?.UtcDateTime,
+                    ban.Reason,
+                    ban.BanningAdmin == null
+                        ? null
+                        : (await _playerLocator.LookupIdAsync(ban.BanningAdmin.Value))?.Username,
+                    unban
+                ));
+            }
```
Comment (ShadowCommander) [inline Content.Server/Administration/BanList/BanListEui.cs:101]: I suggest combining these and instead making an `address` and `hwid` variable that gets set inside the Pii flag check if.

File: Content.Server/Administration/BanList/BanListEui.cs [hunk 2]
```diff
@@ -96,22 +115,42 @@ private async Task LoadRoleBans(NetUserId userId)
...
+                    ban.BanTime.UtcDateTime,
+                    ban.ExpirationTime?.UtcDateTime,
+                    ban.Reason,
+                    ban.BanningAdmin == null
+                        ? null
+                        : (await _playerLocator.LookupIdAsync(ban.BanningAdmin.Value))?.Username,
+                    unban,
+                    ban.Role
+                ));
+            }
```
Comment (ShadowCommander) [inline Content.Server/Administration/BanList/BanListEui.cs:153]: Same here.

**Review by ShadowCommander — CHANGES_REQUESTED** [review]

File: Content.Shared/Administration/AdminFlags.cs [hunk 3]
```diff
@@ -104,6 +104,11 @@ public enum AdminFlags : uint
 		///</summary>
 		Adminchat = 1 << 17,
 
+        ///<summary>
+        /// Allows you to use Admin chat
```
Comment (ShadowCommander) [inline Content.Shared/Administration/AdminFlags.cs:108]: Needs a description that isn't copy pasted.

**Review by ShadowCommander — CHANGES_REQUESTED** [review]

File: Content.Shared/Administration/AdminFlags.cs (hunk 3 above)
Comment (Geekyhobo) [reply Content.Shared/Administration/AdminFlags.cs:108]: 💀

Comment (Geekyhobo) [reply Content.Shared/Administration/AdminFlags.cs]: done

File: Content.Server/Administration/BanList/BanListEui.cs (hunk 1 above)
Comment (Geekyhobo) [reply Content.Server/Administration/BanList/BanListEui.cs]: done

File: Content.Server/Administration/BanList/BanListEui.cs (hunk 2 above)
Comment (Geekyhobo) [reply Content.Server/Administration/BanList/BanListEui.cs]: done

**Review by ShadowCommander — APPROVED** [review]

**Pull request merged**